File: WeekEndRayTrace/src/Ray.java

public class Ray {
	public final Vec3 origin;
	public final Vec3 direction;

	public Ray(Vec3 origin, Vec3 direction) {
		this.origin = origin;
		this.direction = direction;
	}

	public Vec3 positoinAt(double t) {
		return origin.add(direction.mul(t));
	}
}

File: KotRay/src/kotray/Ray.kt
package kotray

/**
 * @param origin 原点
 * @param direction 方向
 */
data class Ray(val origin: Vec3, val direction: Vec3) {
    /**
     * @param t :
     * @return tにおける位置
     */
    fun positionAt(t: Double): Vec3 = origin + direction * t
}
